src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -57,8 +66,29 @@ async Task<CompletableSynchronizedStorageSession> AdaptOrOpenNewSynchronizedStor
                    ?? await synchronizedStorage.OpenSession(contextBag).ConfigureAwait(false);
         }
 
+        static void LogHandlersInvocation(IIncomingLogicalMessageContext context, List<MessageHandler> handlersToInvoke)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"Processing message type: {context.Message.MessageType}");
+            builder.AppendLine($"Headers: {string.Join(", ", context.Headers.Select(h => $"{h.Key}:{h.Value}").ToArray())}");
+            builder.AppendLine($"Message: {context.Message.Instance}");
+            builder.Append("Handlers to invoke: ");
+
+            foreach (var messageHandler in handlersToInvoke)
+            {
+                builder.Append($"{messageHandler.HandlerType.FullName} => ");
+            }
+
+            builder.Length -= 4;
+
+            logger.Debug(builder.ToString());
+        }
+
+
         readonly ISynchronizedStorageAdapter adapter;
         readonly ISynchronizedStorage synchronizedStorage;
         readonly MessageHandlerRegistry messageHandlerRegistry;
+        static readonly ILog logger = LogManager.GetLogger<RepeatedFailuresOverTimeCircuitBreaker>();
+        static readonly bool isDebugIsEnabled = logger.IsDebugEnabled;
```

Comment:
I don't think we need to cache this value

Reply:
This is actually useful in that the JIT can use a static value to completely remove the code in the `if (isDebugIsEnabled)` check when it's false.

Reply:
> I don't think we need to cache this value

This is what Core already does 🙂
Plus what @bording said (I'll pretend like I knew that 😃)